### PR TITLE
WIP: Final fixes vtk plotting

### PR DESCRIPTION
**Pull request by Punzo**
@lassoan. This is still a in working progress.

I did:

1) addressed the issues reported by you in https://github.com/Slicer/Slicer/pull/795
2) reworked the PlotViewController GUI: now all the properties of the PlotDataNodes are all at the second line and they apply at the current selected nodes. Add/Remove/Rename action to the ComboBox have been added. 
3) Edit action at the PlotDataNodeComboBox is under work: the idea is that clicking the action will open the ViewController Module with widget to set the following properties for the selected PlotDataNode:
  // 1) Set vtkTable
  // 2) Set X-Axis
  // 3) Set Y-Axis
  // 4) Set color plot
  // 5) Set Type
  // 6) Set Markers

tomorrow I think I'll finish, the  we can further discuss or chat about it on an hangout call (next developer hangout or we can set a meeting).

### CODE REVIEW

**Comment by lassoan** [comment]
Thanks Davide, these are all great improvements.

A few questions:
- I managed to move one point by hitting 's' key until it allows me to drag points, but as I remember you selected and moved multiple points at once. Is that feature still available?
- In View controllers module I don't see additional plotting options in Advanced section. Is this expected?
- Do you plan to do further changes in this branch or this is ready for integration?

**Comment by Punzo** [comment]
1) I removed the multi-points click and drag. It was very bugged for many reason. We can add a point in the documentation for this (in future plans).
2) No No, I am still working on the advanced GUI. I am fixing few final annoying bugs
3) I think within today I'll push everything

**Comment by Punzo** [comment]
Hi @lassoan I have finished. The commits were a mess to I squashed all the last modifications to the last (previous) commit. I tested on Qt5/VTK8. In the weekend I'll try to test on Qt4. If you have an installation with Qt4/VTK7 may you try it, please?

**Comment by Punzo** [comment]
Hi @lassoan. I test also on qt4 (but still vtk8) and for me it's good to go. Please, when you have time, feel free to review. Thanks!

**Comment by Punzo** [comment]
screenshot from 2017-10-02 15-30-41

**Comment by lassoan** [comment]
I'll try to review and integrate this today.

**Comment by Punzo** [comment]
screenshot from 2017-10-02 15-30-41

**Comment by Punzo** [comment]
Ok Thanks!! I uploaded a screenshot showing the advanced GUI. In addition, in the PlotViewController  there are the X-Axis, Type and MarkerStyle comboboxes (on the right of the PlotDataController Box). If the value is custom, the values in the advanced GUI will be used. If other values are chosen, then the GUI for that properties in the advanced GUI will be disabled and the value of the properties in the SliceController will be used for all the PlotDataNodes "active" for the current PlotChartNode.

**Comment by Punzo** [comment]
Please let me know also if the tooltip are clear.

File: Modules/Loadable/Colors/Resources/UI/qSlicerColorsModuleWidget.ui [hunk 1]
```diff
@@ -44,6 +44,18 @@
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="qMRMLColorTableComboBox" name="ColorTableComboBox">
+          <property name="enabled">
```
Comment (lassoan) [inline Modules/Loadable/Colors/Resources/UI/qSlicerColorsModuleWidget.ui]: @Punzo Have you made changes to qSlicerColorsModuleWidget.ui intentionally?

**Comment by lassoan** [comment]
Committed r26410

Thanks Davide! There are a couple of GUI issues but they can be fixed in further commits. I'll fix them or report on Mantis.

**Comment by Punzo** [comment]
OK perfect, thanks!! Yes please report on Mantis and later on I'll work on them.

File: Modules/Loadable/Colors/Resources/UI/qSlicerColorsModuleWidget.ui (hunk 1 above)
Comment (Punzo) [reply Modules/Loadable/Colors/Resources/UI/qSlicerColorsModuleWidget.ui]: No sorry didn't see.

Comment (Punzo) [reply Modules/Loadable/Colors/Resources/UI/qSlicerColorsModuleWidget.ui]: ok perfect you have already reverted them. Thanks!

**Comment by lassoan** [comment]
@Punzo you have tested on Linux and I tested on Windows and assumed that it works on Mac as well, but unfortunately there were build errors. Could you give it a try to fix it? After that we can ask someone with a Mac to test it.

See errors here:
http://trunk.cdash.org/viewBuildError.php?onlydeltap&buildid=1105635

**Comment by Punzo** [comment]
I think it is the same issue both on Mac and Linux (c++11 features).  In fact, I see that std::to_string and auto are c++11 only. For linux I could tested only for qt5/VTK8 and qt4/VTK8, which both uses c++11. I tried to compile qt4/VTK7, but I got a lot of errors in ubuntu 17.10. I'll fix these now, but I can't do any tests for qt4/VTK7.

**Pull request closed without merging**